yt_dlp/extractor/bilibili.py

```diff
@@ -502,39 +503,135 @@ def _real_extract(self, url):
             season_info.get('bangumi_title'), season_info.get('evaluate'))
 
 
-class BilibiliChannelIE(InfoExtractor):
-    _VALID_URL = r'https?://space.bilibili\.com/(?P<id>\d+)'
-    _API_URL = "https://api.bilibili.com/x/space/arc/search?mid=%s&pn=%d&jsonp=jsonp"
+def bilibili_space_extract_playlist(fetch_page, get_metadata, get_entries):
+    first_page = fetch_page(1)
+    metadata = get_metadata(first_page)
+
+    paged_list = InAdvancePagedList(
+        lambda idx: get_entries(fetch_page(idx) if idx else first_page),
+        metadata['page_count'], metadata['page_size'])
+
+    return metadata, paged_list
+
+
+class BilibiliSpaceVideoIE(InfoExtractor):
+    _VALID_URL = r'https?://space\.bilibili\.com/(?P<id>\d+)(?P<video>/video)?$'
     _TESTS = [{
         'url': 'https://space.bilibili.com/3985676/video',
-        'info_dict': {},
-        'playlist_mincount': 112,
+        'info_dict': {
+            'id': '3985676',
+        },
+        'playlist_mincount': 178,
     }]
 
-    def _entries(self, list_id):
-        count, max_count = 0, None
+    def _real_extract(self, url):
+        mobj = self._match_valid_url(url)
+        if not mobj.group('video'):
+            self.to_screen(f'{url} may have multiple media types. This run will only download videos.\n'
+                           + f'For example, use {url}/audio for audios.')
+
+        playlist_id = self._match_id(url)
+
+        def fetch_page(page_idx):
+            return self._download_json(
+                'https://api.bilibili.com/x/space/arc/search', playlist_id,
+                note=f'Downloading page {page_idx}',
+                query={'mid': playlist_id, 'pn': page_idx, 'jsonp': 'jsonp'})['data']
+
+        def get_metadata(page_data):
+            page_size = page_data['page']['ps']
+            entry_count = page_data['page']['count']
+            return {
+                'page_count': math.ceil(entry_count / page_size),
+                'page_size': page_size,
+                'entry_count': entry_count,
+                'title': None
+            }
 
-        for page_num in itertools.count(1):
-            data = self._download_json(
-                self._API_URL % (list_id, page_num), list_id, note=f'Downloading page {page_num}')['data']
+        def get_entries(page_data):
+            for entry in traverse_obj(page_data, ('list', 'vlist')) or []:
+                yield self.url_result(f'https://www.bilibili.com/video/{entry["bvid"]}',
+                                      BiliBiliIE.ie_key(), entry['bvid'])
```

Comment:
ie_key() isn't needed

```suggestion
                yield self.url_result(f'https://www.bilibili.com/video/{entry["bvid"]}', BiliBiliIE, entry['bvid'])
```